### ISSUE TITLE
Backward does not work when an intermediate tensor is 0-d tensor

### ISSUE THREAD

**Issue by m1pro** [johnmarktaylor91/torchlens#36]
For example when using \<tensor\>.mean()

```python
import torch
from torch import nn, Tensor
import torchlens as tl


torch.manual_seed(42)


class DebugBlock(nn.Module):
    def __init__(self):
        super().__init__()
        self.linear = nn.Linear(8, 6)

    def forward(self, x: Tensor) -> Tensor:
        h1 = self.linear(x)
        h2 = h1.mean()
        out = (h2 + 3.0) ** 2
        return out


x = torch.randn(10, 8, requires_grad=True)
model = DebugBlock()
model_history = tl.log_forward_pass(model, x, detach_saved_tensors=False, save_gradients=True)
out = model_history[-1].tensor_contents
out.backward(torch.randn_like(out))              # <<<< BUG from here ...
```

**Comment by m1pro** [comment]
Thanks for the prompt fix!

**Comment by johnmarktaylor91** [comment]
Of course! Just curious, what is your use case for inspecting the gradients with TorchLens? I hope it's helpful, just trying to understand user needs.

**Comment by m1pro** [comment]
Inspection of gradients is a standard procedure when working with NN, nothing new beyond that. Just playing with TorchLens to see if it could be a more convenient alternative than custom hooking.
As I get more familiar with the capabilities I might suggest more use cases. Looks promising so far.


p.s.
I recommend you add the above code or similar version of it as a test case for your test suite as well.

**Comment by johnmarktaylor91** [comment]
Thanks for the feedback and yes I was planning to do that. Please feel free to let me know if you have any further suggestions or wishlist items :]
